 - make the findawk function more flexible and prefer gawk over all others
 - fix computeTypeFromMimeType so that it sets the proper var

--- converter/other/anytopnm
+++ converter/other/anytopnm
@@ -21,13 +21,13 @@
 # systems, 'awk' does not have gsub().  On other systems, 'awk' has all the
 # functions and 'nawk' does not exist.  On Solaris 8 and 9, we need 'nawk.
 
-  nawk 1>/dev/null 2>/dev/null 1</dev/null
-    
-  if [ $? -eq 127 ]; then
-    AWK=awk
-  else
-    AWK=nawk
-  fi
+  for AWK in gawk nawk awk ; do
+    $AWK 1>/dev/null 2>/dev/null 1</dev/null
+
+    if [ $? -ne 127 ]; then
+      break
+    fi
+  done
 }
 
 
@@ -90,26 +90,26 @@
     case "$1" in
 
         image/jpeg )
-            type="jfif"
+            filetype="jfif"
             ;;
         image/gif )
-            type="gif"
+            filetype="gif"
             ;;
         image/tiff )
-            type="tiff"
+            filetype="tiff"
             ;;
         image/bmp )
-            type="bmp"
+            filetype="bmp"
             ;;
         image/png )
-            type="png"
+            filetype="png"
             ;;
         image/x-portable-bitmap | image/x-portable-pixmap | \
                 image/x-portable-greymap)
-            type="pnm"
+            filetype="pnm"
             ;;
         image/x-xpm )
-            type="xpm"
+            filetype="xpm"
             ;;
         * )
             filetype=unknown
